### PR TITLE
label command

### PR DESCRIPTION
**Pull request by jcdevs**
Players may `label` items with a custom keyword that only they can see and only they can use.

Caveat: equipment list does not display the keyword, however you can still use it e.g. with `remove`.

### CODE REVIEW

**Comment by mitchej123** [comment]
Looks fine after the conflicts are resolved

**Comment by jcdevs** [comment]
conflicts resolved

File: objects/objectAttr.cpp [hunk 1]
```diff
@@ -129,3 +136,11 @@ bool Object::isLightArmor() const {
 bool Object::isBroken() const {
     return(shotsCur == 0 && shotsMax >= 0 && (type == ObjectType::ARMOR || type == ObjectType::KEY || type == ObjectType::WEAPON));
 }
+
+bool Object::isLabeledBy(const Creature* creature) const {
+    return creature->getId() == label.playerId;
+}
+
+bool Object::isLabelMatch(std::string str) const {
```
Comment (mitchej123) [inline objects/objectAttr.cpp:144]: This should be `const std::string &` so it doesn't copy every time it's invoked

File: include/mudObjects/objects.hpp [hunk 2]
```diff
@@ -403,7 +412,8 @@ class Object: public Container, public Containable {
     [[nodiscard]] bool isLightArmor() const;
     [[nodiscard]] bool needsTwoHands() const;
     [[nodiscard]] bool isQuestValid() const; // Is this object valid for a quest?
-
+    [[nodiscard]] bool isLabeledBy(const std::shared_ptr<Creature> & creature) const; // checks if item label was created by player
+    [[nodiscard]] bool isLabelMatch(std::string str) const; // checks if keyword matches label
```
Comment (jcdevs) [inline include/mudObjects/objects.hpp:416]:
```suggestion
    [[nodiscard]] bool isLabelMatch(const std::string & str) const; // checks if keyword matches label
```

File: objects/objectAttr.cpp (hunk 1 above)
Comment (jcdevs) [inline objects/objectAttr.cpp:144]:
```suggestion
bool Object::isLabelMatch(const std::string & str) const {
```

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**